server/resource_server/service/service_test.go

```diff
@@ -64,8 +96,18 @@ func (m *mockRepository) ProfileUser(userID uint) (models.User, []models.UserMet
 	return models.User{}, []models.UserMetadata{}, fmt.Errorf("User not found")
 }
 
-func (m *mockRepository) VerifyEmail(userID uint) error {
-	return nil
+func (m *mockRepository) SaveProofOfEmailVerification(
+	userID uint, verificationCode string, proof string,
+) error {
+	return m.saveProofOfEmailVerification(userID, verificationCode, proof)
+}
+
+func (m *mockRepository) SaveEmailVerificationData(data models.EmailVerificationData) error {
+	return m.saveEmailVerificationData(data)
+}
+
+func (m *mockRepository) GetEmailVerificationData(userId uint) (models.EmailVerificationData, error) {
```

Comment:
Oh... I see look s like you have a pattern going. When a mock method is to be overwritten, you put a placeholder recursive implementation as the default case.